samza-sql/src/main/java/org/apache/samza/sql/data/SamzaSqlRelMessage.java

```diff
@@ -56,9 +64,7 @@ public SamzaSqlRelMessage(Object key, List<String> names, List<Object> values) {
     this.key = key;
     this.value.addAll(values);
     this.names.addAll(names);
-    if (key != null) {
-      this.relFieldValues.add(key);
-    }
+    this.relFieldValues.add(key);
```

Comment:
question: it is ok to add a null key to a message?

nitpick: can you add comment to the code (maybe at the getter level) of what is  the difference between FieldValues() and RelFieldValues(). Seems like the latter is just  values array prefixed by the key value ([key] + values). Why we need both?